bpf/lib/node.h

```diff
@@ -65,3 +65,45 @@ lookup_ip6_node_id(const union v6addr *ip6)
 	return node_value->id;
 }
 # endif /* ENABLE_IPV6 */
+
+struct {
+	__uint(type, BPF_MAP_TYPE_HASH);
+	__type(key, struct device_key);
+	__type(value, struct device_value);
+	__uint(pinning, LIBBPF_PIN_BY_NAME);
+	__uint(max_entries, DEVICE_MAP_SIZE);
+	__uint(map_flags, BPF_F_NO_PREALLOC);
+} cilium_device_map  __section_maps_btf;
+
+static __always_inline struct device_value *
+lookup_device(__u32 ifindex)
+{
+	struct device_key key = {};
+
+	key.ifindex = ifindex;
+
+	return map_lookup_elem(&cilium_device_map, &key);
+}
+
+static __always_inline bool
+is_l3_device(__u32 ifindex)
+{
+	struct device_value *device_value;
+
+	device_value = lookup_device(ifindex);
+	if (!device_value)
+		return false;
+	return device_value->l3 > 0;
+}
+
+static __always_inline union macaddr
+device_mac(__u32 ifindex)
+{
+	union macaddr dev_mac = {0};
+	struct device_value *device_value;
+
+	device_value = lookup_device(ifindex);
+	if (!device_value)
+		return dev_mac;
+	return device_value->mac;
+}
```

Comment:
`node.h` wraps the node map, which contains *remote* nodes. But now we want to manage network interfaces on the _local_ node. So I think this would be best placed into a separate header file - this could be named `network_devices.h` ?

Reply:
I have moved this code to a new header file `network_device.h`. Please take a look.